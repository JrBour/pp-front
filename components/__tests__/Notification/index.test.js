import { shallowMount } from '@vue/test-utils'
import Notification from '@/components/Notification'

const event = {
  id: '12',
  startAt: '12/03/2021',
  name: 'Event name Lorem ipsum Lorem ipsum',
  address: '12 rue du gnou',
  city: 'Paris'
}

describe('Notification', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Notification, {
      propsData: {
        event
      }
    })
  })

  describe('method', () => {
    it('should emit submit-response method', async () => {
      const acceptButton = wrapper.findAll('button').at(0)
      await acceptButton.trigger('click')
      expect(wrapper.emitted('submit-response')).toBeTruthy()
    })
  })

  describe('snapshot', () => {
    it('renders properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
