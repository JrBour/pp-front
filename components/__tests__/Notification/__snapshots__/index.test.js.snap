// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification snapshot renders properly 1`] = `
<div class="notification__wrapper">
  <div class="notifications__date">
    <p class="notifications__date_day">Ven</p>
    <p class="notifications__date_date">3</p>
    <p class="notifications__date_month">
      Dec
    </p>
  </div>
  <div class="notifications__separator"></div>
  <div class="notifications__informations">
    <h2>
      Event name Lore...
    </h2>
    <p>12 rue du gnou - Paris</p>
  </div>
  <div class="notifications__actions"><button class="notifications__accept"><img src=""></button> <button class="notifications__deny"><img src=""></button></div>
</div>
`;
